diff -Naur p4-examples-9.11.0.orig/p4_16_programs/tna_mirror/test.py p4-examples-9.11.0/p4_16_programs/tna_mirror/test.py
--- p4-examples-9.11.0.orig/p4_16_programs/tna_mirror/test.py	2022-10-26 10:59:46.000000000 +0200
+++ p4-examples-9.11.0/p4_16_programs/tna_mirror/test.py	2024-04-12 13:08:29.773029877 +0200
@@ -164,7 +164,7 @@
         duplicate_add_error = False
         duplicate_add_error_p4_table = False
         try:
-            ports = random.sample(set(swports), len(swports)//2)
+            ports = random.sample(sorted(set(swports)), len(swports)//2)
             logger.info("swports: %s", swports)
             logger.info("selected ports: %s ", ports)
             sid = BASE_SID_NORM
@@ -250,7 +250,7 @@
         try:
             # programming the multicast group
             # Group a will use at most half of the swports
-            ports_a = random.sample(set(swports), len(swports)//2)
+            ports_a = random.sample(sorted(set(swports)), len(swports)//2)
             logger.info("swports: %s ", swports)
             logger.info("ports_a: %s ", ports_a)
 
diff -Naur p4-examples-9.11.0.orig/ptf-tests/iterator/test.py p4-examples-9.11.0/ptf-tests/iterator/test.py
--- p4-examples-9.11.0.orig/ptf-tests/iterator/test.py	2022-10-26 10:59:47.000000000 +0200
+++ p4-examples-9.11.0/ptf-tests/iterator/test.py	2024-04-12 13:09:10.429214337 +0200
@@ -941,7 +941,7 @@
             all_keys = list(range(30))
             to_mod = random.sample(all_keys, 15)
             to_del = random.sample(to_mod, 5)
-            to_del += random.sample(set(all_keys) - set(to_mod), 5)
+            to_del += random.sample(sorted(set(all_keys) - set(to_mod)), 5)
 
             # Issue the adds, mods, and deletes to the virtual device.
             key_to_ms = dict()
