diff -Naur bf-platforms-9.8.0-orig/drivers/src/bf_port_mgmt/bf_pm_qsfp_mgmt.c bf-platforms-9.8.0/drivers/src/bf_port_mgmt/bf_pm_qsfp_mgmt.c
--- bf-platforms-9.8.0-orig/drivers/src/bf_port_mgmt/bf_pm_qsfp_mgmt.c	2022-01-19 08:14:37.000000000 +0100
+++ bf-platforms-9.8.0/drivers/src/bf_port_mgmt/bf_pm_qsfp_mgmt.c	2023-10-24 14:36:32.004994915 +0200
@@ -265,6 +265,7 @@
   int bitnum;          // bit number to log
   Sff_flag_type type;  // event or status
   char desc[50];
+  bool media_lane;     // true if flag is per media lane
   int fval_prev[BF_PLAT_MAX_QSFP + 1];
   int fval_cur[BF_PLAT_MAX_QSFP + 1];
 } qsfp_flag_log_t;
@@ -342,10 +343,10 @@
 // this array should only contain flags that are logged
 qsfp_flag_log_t ln_flag_log[] = {
     {
-     FLAG_TX_LOS, FLAG_INFO, 0, FLAG_TYPE_STATUS, "FLAG_TX_LOS",
+      FLAG_TX_LOS, FLAG_INFO, 0, FLAG_TYPE_STATUS, "FLAG_TX_LOS", false,
     },
     {
-     FLAG_RX_LOS, FLAG_INFO, 0, FLAG_TYPE_STATUS, "FLAG_RX_LOS",
+      FLAG_RX_LOS, FLAG_INFO, 0, FLAG_TYPE_STATUS, "FLAG_RX_LOS", true,
     },
     {
      FLAG_TX_ADAPT_EQ_FAULT,
@@ -353,15 +354,16 @@
      0,
      FLAG_TYPE_EVENT,
      "FLAG_TX_ADAPT_EQ_FAULT",
+     false,
     },
     {
-     FLAG_TX_FAULT, FLAG_ERROR, 0, FLAG_TYPE_EVENT, "FLAG_TX_FAULT",
+      FLAG_TX_FAULT, FLAG_ERROR, 0, FLAG_TYPE_EVENT, "FLAG_TX_FAULT", true,
     },
     {
-     FLAG_TX_LOL, FLAG_INFO, 0, FLAG_TYPE_STATUS, "FLAG_TX_LOL",
+      FLAG_TX_LOL, FLAG_INFO, 0, FLAG_TYPE_STATUS, "FLAG_TX_LOL", false,
     },
     {
-     FLAG_RX_LOL, FLAG_INFO, 0, FLAG_TYPE_STATUS, "FLAG_RX_LOL",
+      FLAG_RX_LOL, FLAG_INFO, 0, FLAG_TYPE_STATUS, "FLAG_RX_LOL", true,
     },
     {
      FLAG_RX_PWR_HIGH_ALARM,
@@ -369,6 +371,7 @@
      0,
      FLAG_TYPE_STATUS,
      "FLAG_RX_PWR_HIGH_ALARM",
+     true,
     },
     {
      FLAG_RX_PWR_LOW_ALARM,
@@ -376,6 +379,7 @@
      0,
      FLAG_TYPE_STATUS,
      "FLAG_RX_PWR_LOW_ALARM",
+     true,
     },
     {
      FLAG_TX_BIAS_HIGH_ALARM,
@@ -383,6 +387,7 @@
      0,
      FLAG_TYPE_STATUS,
      "FLAG_TX_BIAS_HIGH_ALARM",
+     true,
     },
     {
      FLAG_TX_BIAS_LOW_ALARM,
@@ -390,6 +395,7 @@
      0,
      FLAG_TYPE_STATUS,
      "FLAG_TX_BIAS_LOW_ALARM",
+     true,
     },
     {
      FLAG_TX_PWR_HIGH_ALARM,
@@ -397,6 +403,7 @@
      0,
      FLAG_TYPE_STATUS,
      "FLAG_TX_PWR_HIGH_ALARM",
+     true,
     },
     {
      FLAG_TX_PWR_LOW_ALARM,
@@ -404,6 +411,7 @@
      0,
      FLAG_TYPE_STATUS,
      "FLAG_TX_PWR_LOW_ALARM",
+     true,
     },
 };
 
@@ -1309,7 +1317,8 @@
 static void qsfp_log_lane_flags(int conn_id) {
   int loopcnt, loopmax, flag_val, ln;
   bool logflag;
-
+  dev_cfg_per_channel_t *dev_cfg;
+  
   loopmax = sizeof(ln_flag_log) / sizeof(qsfp_flag_log_t);
 
   // get current flag values
@@ -1324,9 +1333,28 @@
     ln_flag_log[loopcnt].fval_cur[conn_id] = 0;
 
     for (ln = 0; ln < qsfp_state[conn_id].ch_cnt; ln++) {
-      // get the current flag value for each lane from the cache
-      flag_val =
+      // get the current flag value for each lane from the cache. we
+      // iterate over the host lanes but some of the flags are per
+      // media-lane. this should probably be handled better at a more
+      // fundamental level. as a workaround, we use the
+      // host-lane-to-media-lane mapping that is established when the
+      // application is selected. this happens only after the port is
+      // enabled.
+      if (ln_flag_log[loopcnt].media_lane) {
+        // media_ch is set to -1 after port-add but before
+        // port-enb. also check for host_intf_lanes to to cover the
+        // case when the port hasn't been added yet.
+        dev_cfg = &qsfp_state[conn_id].dev_cfg[ln];
+        if (dev_cfg->host_intf_nlanes > 0 && dev_cfg->media_ch != -1) {
+          flag_val =
+            bf_qsfp_get_flag(conn_id, ln_flag_log[loopcnt].flag, dev_cfg->media_ch, 0, NULL);
+        } else {
+          flag_val = 0;
+        }
+      } else {
+        flag_val =
           bf_qsfp_get_flag(conn_id, ln_flag_log[loopcnt].flag, ln, 0, NULL);
+      }
 
       // if set, add to bitmask for this flag
       if (flag_val >= 0) {
