diff -Naur bf-platforms-9.9.0.orig/CMakeLists.txt bf-platforms-9.9.0/CMakeLists.txt
--- bf-platforms-9.9.0.orig/CMakeLists.txt	2022-07-13 12:34:15.000000000 +0200
+++ bf-platforms-9.9.0/CMakeLists.txt	2022-07-14 10:24:33.152150896 +0200
@@ -29,9 +29,6 @@
     set(PLATFORM "" CACHE STRING "Name of APS Networks platform to build")
     set(SDE_INSTALL "" CACHE PATH "Path to SDE install location")
 
-    if (SDE_INSTALL STREQUAL "")
-        message(FATAL_ERROR "SDE_INSTALL is not set")
-    endif()
     if (PLATFORM STREQUAL "")
         message(FATAL_ERROR "PLATFORM is not set")
     endif()
