diff -Naur bf-drivers-9.6.0.orig/src/bf_rt/CMakeLists.txt bf-drivers-9.6.0/src/bf_rt/CMakeLists.txt
--- bf-drivers-9.6.0.orig/src/bf_rt/CMakeLists.txt	2021-07-14 06:32:39.000000000 +0200
+++ bf-drivers-9.6.0/src/bf_rt/CMakeLists.txt	2021-07-26 10:19:20.791164434 +0200
@@ -78,8 +78,9 @@
 )
 
 if(BF-PYTHON)
+  find_library(BF_PYTHON_LIBRARY NAMES python3.4m libpython3.4m)
   add_library(bfpythonlib UNKNOWN IMPORTED)
-  set_property(TARGET bfpythonlib PROPERTY IMPORTED_LOCATION ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libpython3.4m.so)
+  set_property(TARGET bfpythonlib PROPERTY IMPORTED_LOCATION ${BF_PYTHON_LIBRARY})
   add_dependencies(bfpythonlib libpython3.4m)
   add_library(bfshell_plugin_bf_rt_o OBJECT cli/bf_rt_cli.c)
   add_dependencies(bfshell_plugin_bf_rt_o bfpythonlib)
diff -Naur bf-drivers-9.6.0.orig/src/lld/CMakeLists.txt bf-drivers-9.6.0/src/lld/CMakeLists.txt
--- bf-drivers-9.6.0.orig/src/lld/CMakeLists.txt	2021-07-14 06:32:39.000000000 +0200
+++ bf-drivers-9.6.0/src/lld/CMakeLists.txt	2021-07-26 10:21:45.195162958 +0200
@@ -61,8 +61,9 @@
 add_library(lld SHARED EXCLUDE_FROM_ALL $<TARGET_OBJECTS:lld_o>)
 
 if(BF-PYTHON)
+  find_library(BF_PYTHON_LIBRARY NAMES python3.4m libpython3.4m)
   add_library(bfpythonlib UNKNOWN IMPORTED)
-  set_property(TARGET bfpythonlib PROPERTY IMPORTED_LOCATION ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libpython3.4m.so)
+  set_property(TARGET bfpythonlib PROPERTY IMPORTED_LOCATION ${BF_PYTHON_LIBRARY})
   add_dependencies(bfpythonlib libpython3.4m)
   add_library(bfshell_plugin_debug_o OBJECT cli/debug_cli.c)
   add_dependencies(bfshell_plugin_debug_o bfpythonlib)
