diff -Naur bf-drivers-9.8.0.orig/src/bf_rt/CMakeLists.txt bf-drivers-9.8.0/src/bf_rt/CMakeLists.txt
--- bf-drivers-9.8.0.orig/src/bf_rt/CMakeLists.txt	2022-01-19 08:03:21.000000000 +0100
+++ bf-drivers-9.8.0/src/bf_rt/CMakeLists.txt	2022-02-07 10:36:20.426086638 +0100
@@ -79,8 +79,9 @@
 )
 
 if(BF-PYTHON)
+  find_library(BF_PYTHON_LIBRARY NAMES python3.8 libpython3.8)
   add_library(bfpythonlib UNKNOWN IMPORTED)
-  set_property(TARGET bfpythonlib PROPERTY IMPORTED_LOCATION ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libpython${BF_PYTHON_VER}.so)
+  set_property(TARGET bfpythonlib PROPERTY IMPORTED_LOCATION ${BF_PYTHON_LIBRARY})
   add_dependencies(bfpythonlib libpython${BF_PYTHON_VER})
   add_library(bfshell_plugin_bf_rt_o OBJECT cli/bf_rt_cli.c)
   include_directories(${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/../include/python${BF_PYTHON_VER})
diff -Naur bf-drivers-9.8.0.orig/src/lld/CMakeLists.txt bf-drivers-9.8.0/src/lld/CMakeLists.txt
--- bf-drivers-9.8.0.orig/src/lld/CMakeLists.txt	2022-01-19 08:02:34.000000000 +0100
+++ bf-drivers-9.8.0/src/lld/CMakeLists.txt	2022-02-07 10:36:15.818086685 +0100
@@ -73,8 +73,9 @@
 add_library(lld SHARED EXCLUDE_FROM_ALL $<TARGET_OBJECTS:lld_o>)
 
 if(BF-PYTHON)
+  find_library(BF_PYTHON_LIBRARY NAMES python3.8 libpython3.8)
   add_library(bfpythonlib UNKNOWN IMPORTED)
-  set_property(TARGET bfpythonlib PROPERTY IMPORTED_LOCATION ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libpython${BF_PYTHON_VER}.so)
+  set_property(TARGET bfpythonlib PROPERTY IMPORTED_LOCATION ${BF_PYTHON_LIBRARY})
   add_dependencies(bfpythonlib libpython${BF_PYTHON_VER})
   add_library(bfshell_plugin_debug_o OBJECT cli/debug_cli.c)
   add_dependencies(bfshell_plugin_debug_o bfpythonlib)
