diff -Naur ptf-modules-9.6.0.orig/bf-ptf/setup.py ptf-modules-9.6.0/bf-ptf/setup.py
--- ptf-modules-9.6.0.orig/bf-ptf/setup.py	2021-07-14 06:43:03.000000000 +0200
+++ ptf-modules-9.6.0/bf-ptf/setup.py	2021-07-15 13:37:45.724760489 +0200
@@ -16,8 +16,7 @@
     ],
     package_dir={'': 'src'},
     scripts=[
-        'ptf',
-        'ptf_nn/ptf_nn_agent.py'
+        'ptf'
     ],
     zip_safe=False,
     license='Apache License',
