diff -Naaur ptf-modules-9.1.1.orig/ptf-utils/veth_setup.sh ptf-modules-9.1.1/ptf-utils/veth_setup.sh
--- ptf-modules-9.1.1.orig/ptf-utils/veth_setup.sh	2020-03-24 21:14:26.000000000 +0100
+++ ptf-modules-9.1.1/ptf-utils/veth_setup.sh	2022-08-16 14:39:51.553267583 +0200
@@ -22,8 +22,8 @@
     ip link set dev $intf1 mtu 10240 up
     TOE_OPTIONS="rx tx sg tso ufo gso gro lro rxvlan txvlan rxhash"
     for TOE_OPTION in $TOE_OPTIONS; do
-       /sbin/ethtool --offload $intf0 "$TOE_OPTION" off &> /dev/null
-       /sbin/ethtool --offload $intf1 "$TOE_OPTION" off &> /dev/null
+       /sbin/ethtool --offload $intf0 "$TOE_OPTION" off &> /dev/null || true
+       /sbin/ethtool --offload $intf1 "$TOE_OPTION" off &> /dev/null || true
     done
     # do not fail if ipv6 is disabled system-wide
     sysctl net.ipv6.conf.$intf0.disable_ipv6=1 &> /dev/null || true
