diff -Naur bf-sde-9.5.0.orig/run_bfshell.sh bf-sde-9.5.0/run_bfshell.sh
--- bf-sde-9.5.0.orig/run_bfshell.sh	2021-04-14 00:58:25.000000000 +0200
+++ bf-sde-9.5.0/run_bfshell.sh	2021-04-30 08:53:02.314189371 +0200
@@ -77,7 +77,7 @@
 fi
 if [ "$TARGET" != "bmv2" ]; then
   if [ $SKIP_STATUS_SRV = false ]; then
-    python $SDE_INSTALL/lib/python2.7/site-packages/p4testutils/bf_switchd_dev_status.py \
+    $SDE_INSTALL/lib/python2.7/site-packages/p4testutils/bf_switchd_dev_status.py \
       $STS_HOST_STR $STS_PORT_STR $STS_DEV_STR
   fi
 fi
