diff -Naur bf-bsp-8.9.x-6686d64c.orig/configure.ac bf-bsp-8.9.x-6686d64c/configure.ac
--- bf-bsp-8.9.x-6686d64c.orig/configure.ac	2023-06-30 11:11:22.000000000 +0200
+++ bf-bsp-8.9.x-6686d64c/configure.ac	2023-07-10 11:05:52.057749362 +0200
@@ -59,6 +59,13 @@
 #AM_CONDITIONAL([BUILD_TOFINO_BRINGUP_PLAT], [test "x$build_tof_brgup" = xyes])
 #AM_CONDITIONAL([BUILD_NEWPORT_PLAT], [test "x$build_newport" = xyes])
 AM_CONDITIONAL([BUILD_ASTERFUSIONBF_PLAT], [test "x$build_asterfusionbf" = xyes])
+AM_CONDITIONAL([BUILD_ASTERFUSIONBF_PLAT], [test "x$build_asterfusionbf" = xyes])
+AC_ARG_WITH([sde_version],
+	[AS_HELP_STRING([--with-sde-version],
+	[SDE version as numeric value (e.g. 990) [default=990]])],
+	[],
+	[with_sde_version=990])
+AM_CONDITIONAL([SDE_AT_LEAST_990],[test ${with_sde_version} -ge 990])
 
 asterfusion_diags=no
 AC_ARG_WITH([asterfusion-diags],
diff -Naur bf-bsp-8.9.x-6686d64c.orig/drivers/include/bf_pltfm_types/bf_pltfm_types.h bf-bsp-8.9.x-6686d64c/drivers/include/bf_pltfm_types/bf_pltfm_types.h
--- bf-bsp-8.9.x-6686d64c.orig/drivers/include/bf_pltfm_types/bf_pltfm_types.h	2023-06-30 11:11:22.000000000 +0200
+++ bf-bsp-8.9.x-6686d64c/drivers/include/bf_pltfm_types/bf_pltfm_types.h	2023-07-10 11:05:52.057749362 +0200
@@ -31,7 +31,9 @@
 /* Mainline SDE version used by bsp, set 9.5.0 as default.
  * Valid value in [891,900,910,930,950,970,990,9110,9120,...].
  * A sub version start from a given mainline is valid too, such as 931,952,971, etc. */
+#ifndef SDE_VERSION
 #define SDE_VERSION 950
+#endif
 #define SDE_VERSION_EQ(key) \
         EQ(SDE_VERSION, (key))
 
@@ -42,7 +44,9 @@
         LT(SDE_VERSION, (key))
 
 /* Mainline OS version, <= 9 or > 9. Valid value in [8,9,10,11]. */
+#ifndef OS_VERSION
 #define OS_VERSION 9
+#endif
 #define OS_VERSION_EQ(key) \
         EQ(OS_VERSION, (key))
 
@@ -103,7 +107,7 @@
 #endif
 
 
-#if SDE_VERSION_LT(980)
+#if SDE_VERSION_LT(990)
 #ifdef INC_PLTFM_UCLI
 #include <bfutils/uCli/ucli.h>
 #include <bfutils/uCli/ucli_argparse.h>
diff -Naur bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/cp2112_util/Makefile.am bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/cp2112_util/Makefile.am
--- bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/cp2112_util/Makefile.am	2023-06-30 11:11:22.000000000 +0200
+++ bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/cp2112_util/Makefile.am	2023-07-10 11:05:52.057749362 +0200
@@ -19,10 +19,15 @@
 # Different SDE on different OS.
 # Default to OS <= 9 and SDE < 9.9.x. by tsihang, 2022-11-09.
 # For SDE version <  9.9.x && OS <= 9
-cp2112_util_LDADD = -lbfsys -lusb-1.0 -lcurl -lcgos
+#cp2112_util_LDADD = -lbfsys -lusb-1.0 -lcurl -lcgos
 # For SDE version >= 9.9.x && OS <= 9
 #cp2112_util_LDADD =  -ltarget_sys -lusb-1.0 -lcurl -lcgos
 # For SDE version <  9.9.x && OS >= 10
 #cp2112_util_LDADD =  -lbfsys -lusb-1.0 -lcurl -lcgos -li2c
 # For SDE version >= 9.9.x && OS >= 10
 #cp2112_util_LDADD =  -ltarget_sys -lusb-1.0 -lcurl -lcgos -li2c
+if SDE_AT_LEAST_990
+cp2112_util_LDADD =  -ltarget_sys -lusb-1.0 -lcurl -lcgos -li2c
+else
+cp2112_util_LDADD =  -lbfsys -lusb-1.0 -lcurl -lcgos -li2c
+endif
diff -Naur bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/src/bf_pltfm_chss_mgmt/bf_pltfm_bd_eeprom.c bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/src/bf_pltfm_chss_mgmt/bf_pltfm_bd_eeprom.c
--- bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/src/bf_pltfm_chss_mgmt/bf_pltfm_bd_eeprom.c	2023-06-30 11:11:22.000000000 +0200
+++ bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/src/bf_pltfm_chss_mgmt/bf_pltfm_bd_eeprom.c	2023-07-10 11:10:59.262433475 +0200
@@ -939,7 +939,7 @@
             /* First boot error, no chance to rescure, exit forcely. */
             exit (0);
         } else {
-            char tlv_str[128];
+            char tlv_str[lqe_valen];
             char tlv_header[128];
             char *p;
             while (fgets (tlv_str, lqe_valen, fp)) {
diff -Naur bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/src/bf_pltfm_master_i2c/bf_pltfm_master_i2c.c bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/src/bf_pltfm_master_i2c/bf_pltfm_master_i2c.c
--- bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/src/bf_pltfm_master_i2c/bf_pltfm_master_i2c.c	2023-06-30 11:11:22.000000000 +0200
+++ bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/src/bf_pltfm_master_i2c/bf_pltfm_master_i2c.c	2023-07-10 11:06:49.109876212 +0200
@@ -579,7 +579,7 @@
     struct tm *tmp_ptr;
     struct stat file_stat;
     char *dbgstr_ptr = debug_str;
-    char rfile[128] = {0};
+    char rfile[512] = {0};
     const char *dbg_file = LOG_DIR_PREFIX"/sio_debug.log";
 
     if (!bf_sio_dbg_enabled ||
diff -Naur bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/src/Makefile.am bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/src/Makefile.am
--- bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/src/Makefile.am	2023-06-30 11:11:22.000000000 +0200
+++ bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/src/Makefile.am	2023-07-10 11:05:52.057749362 +0200
@@ -114,7 +114,7 @@
 libasterfusionbf_driver_la_LIBADD = \
 libasterfusionbf_platform.la
 else
-libasterfusionbf_driver_la_LIBADD = \
+libasterfusionbf_driver_la_LIBADD = -li2c \
 libasterfusionbf_platform.la \
 libasterfusionbf_cp2112.la \
 libasterfusionbf_chss_mgmt.la \
diff -Naur bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/tofino_spi_util/Makefile.am bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/tofino_spi_util/Makefile.am
--- bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/tofino_spi_util/Makefile.am	2023-06-30 11:11:22.000000000 +0200
+++ bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/tofino_spi_util/Makefile.am	2023-07-10 11:05:52.057749362 +0200
@@ -11,7 +11,7 @@
 # Different SDE on different OS.
 # Default to OS <= 9 and SDE < 9.9.x. by tsihang, 2022-11-09.
 # For SDE version <  9.9.x && OS <= 9
-tofino_spi_util_LDADD = -lbfsys -lusb-1.0 -lcurl -lcgos
+#tofino_spi_util_LDADD = -lbfsys -lusb-1.0 -lcurl -lcgos
 # For SDE version >= 9.9.x && OS <= 9
 #tofino_spi_util_LDADD =  -ltarget_sys -ldriver -lusb-1.0 -lcurl -lcgos
 # For SDE version <  9.9.x && OS >= 10
@@ -19,5 +19,11 @@
 # For SDE version >= 9.9.x && OS >= 10
 #tofino_spi_util_LDADD =  -ltarget_sys -ldriver -lusb-1.0 -lcurl -lcgos -li2c
 
+if SDE_AT_LEAST_990
+tofino_spi_util_LDADD =  -ltarget_sys -ldriver -lusb-1.0 -lcurl -lcgos -li2c
+else
+tofino_spi_util_LDADD =  -lbfsys -ldriver -lusb-1.0 -lcurl -lcgos -li2c
+endif
+
 # For SDE version <  9.7.x
 #tofino_spi_util_LDADD += -lbf_switchd_lib
diff -Naur bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/uart_util/Makefile.am bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/uart_util/Makefile.am
--- bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/uart_util/Makefile.am	2023-06-30 11:11:22.000000000 +0200
+++ bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/uart_util/Makefile.am	2023-07-10 11:05:52.057749362 +0200
@@ -17,10 +17,15 @@
 # Different SDE on different OS.
 # Default to OS <= 9 and SDE < 9.9.x. by tsihang, 2022-11-09.
 # For SDE version <  9.9.x && OS <= 9
-uart_util_LDADD = -lbfsys -lusb-1.0 -lcurl -lcgos
+#uart_util_LDADD = -lbfsys -lusb-1.0 -lcurl -lcgos
 # For SDE version >= 9.9.x && OS <= 9
 #uart_util_LDADD =  -ltarget_sys -lusb-1.0 -lcurl -lcgos
 # For SDE version <  9.9.x && OS >= 10
 #uart_util_LDADD =  -lbfsys -lusb-1.0 -lcurl -lcgos -li2c
 # For SDE version >= 9.9.x && OS >= 10
 #uart_util_LDADD =  -ltarget_sys -lusb-1.0 -lcurl -lcgos -li2c
+if SDE_AT_LEAST_990
+uart_util_LDADD =  -ltarget_sys -lusb-1.0 -lcurl -lcgos -li2c
+else
+uart_util_LDADD =  -lbfsys -lusb-1.0 -lcurl -lcgos -li2c
+endif
diff -Naur bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/utils/xt-cfgen.sh bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/utils/xt-cfgen.sh
--- bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/utils/xt-cfgen.sh	2023-06-30 11:11:22.000000000 +0200
+++ bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/utils/xt-cfgen.sh	2023-07-10 11:05:52.057749362 +0200
@@ -17,15 +17,13 @@
 enable_iic=1
 
 echo -e "${YELLOW}Notice: Start detecting and make sure that the switchd is not running${RES}"
-sleep 1
-install_bfnkdrv
 
 # Find uart_util which created by bsp and installed to $SDE_INSTALL/bin.
 # When found return its absolute path.
 # For a given X-T with different OS (such as ubuntu), the working tty may differ from /dev/ttyS1 on Debian.
 # Please help to test with all possible /dev/ttySx.
 # uart_util=`find / | grep uart_util`
-uart_util="$SDE_INSTALL/bin/uart_util"
+uart_util="$(type -p uart_util)"
 if [ ! -e $uart_util ]; then
     # For SONiC, uart_util is installed in /opt/bfn/install/bin
     uart_util="/opt/bfn/install/bin/uart_util"
@@ -312,12 +310,11 @@
 echo "COMe     : $default_cme"
 
 if [ -f $cfgfile ]; then
-    echo ""
-    echo ""
-    echo -e "${RED} We found that the $cfgfile already exists and no new configuration file will be generated.${RES}"
-    echo -e "${RED} The purpose of this prompt is to take into account that your system may have worked fine in the past.${RES}"
-    echo -e "${RED} If you want to generate a new configuration file, delete the old one first.${RES}"
-    friendly_exit
+    if [ -f $cfgfile.lock ]; then
+        echo -e "${YELLOW}$cfgfile exists and is protected by $cfgfile.lock.${RES}"
+	exit 0
+    fi
+    echo -e "${YELLOW}Overwriting existing $cfgfile. Create $cfgfile.lock to prevent this.${RES}"
 fi
 
 echo ""
diff -Naur bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/utils/xt-setup.sh bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/utils/xt-setup.sh
--- bf-bsp-8.9.x-6686d64c.orig/platforms/asterfusion-bf/utils/xt-setup.sh	2023-06-30 11:11:22.000000000 +0200
+++ bf-bsp-8.9.x-6686d64c/platforms/asterfusion-bf/utils/xt-setup.sh	2023-07-10 11:05:52.057749362 +0200
@@ -22,7 +22,7 @@
 
 install_cgosdrv()
 {
-    cgosdrv="/lib/modules/`uname -r`/kernel/drivers/misc/cgosdrv.ko"
+    cgosdrv="${MODULES}/lib/modules/$(uname -r)/cgosdrv.ko"
     mod=`lsmod | grep cgosdrv`
     mod=${mod:0:7}
 
@@ -52,7 +52,7 @@
 
 install_nct6779d()
 {
-    nct6779drv="/lib/modules/`uname -r`/kernel/drivers/misc/nct6779d.ko"
+    nct6779drv="${MODULES}/lib/modules/$(uname -r)/nct6779d.ko"
     mod=`lsmod | grep nct6779d`
     mod=${mod:0:8}
 
@@ -88,7 +88,7 @@
         $SDE_INSTALL/bin/dma_setup.sh
     fi
 
-    bfnkdrv="$SDE_INSTALL/lib/modules/bf_kdrv.ko"
+    bfnkdrv="${MODULES}/lib/modules/bf_kdrv.ko"
     mod=`lsmod | grep bf_kdrv`
     mod=${mod:0:7}
 
