diff --git a/CMakeLists.txt b/CMakeLists.txt
index 238f102..77b2251 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,7 @@
 #
 
 cmake_minimum_required(VERSION 3.5)
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
 option(THRIFT-DRIVER      "Build with thrift support" ON)
 option(LASER_ON           "Build for forcely enabling laser when modules plugged-in" OFF)
@@ -125,7 +126,7 @@ link_directories(AFTER ${CMAKE_INSTALL_PREFIX}/lib)
 include_directories(SYSTEM ${CMAKE_INSTALL_PREFIX}/include)
 
 if (THRIFT-DRIVER)
-  find_package(Thrift REQUIRED)
+  include(FindThrift)
 endif()
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}     -DINC_PLTFM_UCLI -Wno-unused-parameter -Wno-shadow")
@@ -247,9 +248,9 @@ set_target_properties(pltfm_mgr PROPERTIES VERSION 0.0.0 SOVERSION 0)
 #thrift
 if (THRIFT-DRIVER)
   add_subdirectory(platforms/asterfusion-bf/thrift)
-  target_link_libraries(pltfm_mgr PUBLIC pltfm_mgr_thrift ${bfsys_lib} curl usb-1.0 ${private_lib})
+  target_link_libraries(pltfm_mgr PUBLIC pltfm_mgr_thrift ${bfsys_lib} curl usb-1.0 ${private_lib} ${i2c_lib})
 else()
-  target_link_libraries(pltfm_mgr PUBLIC ${bfsys_lib} curl usb-1.0 ${private_lib})
+  target_link_libraries(pltfm_mgr PUBLIC ${bfsys_lib} curl usb-1.0 ${private_lib} ${i2c_lib})
 endif()
 
 add_library(asterfusion_bin_srcs_o OBJECT
diff --git a/drivers/include/bf_pltfm_types/bf_pltfm_types.h b/drivers/include/bf_pltfm_types/bf_pltfm_types.h
index 3206982..3b0c9e2 100644
--- a/drivers/include/bf_pltfm_types/bf_pltfm_types.h
+++ b/drivers/include/bf_pltfm_types/bf_pltfm_types.h
@@ -199,9 +199,9 @@ extern "C" {
             ____ubuf[strlen(____ubuf) - 1] = 0;                                 \
             ____size += snprintf(____info + ____size, AF_LOG_MSG_SIZE,          \
                         "%s %s.%06d ", ____tbuf, ____ubuf, (int)__tm__.tv_usec);\
-            ____size += snprintf(____info + ____size, AF_LOG_MSG_SIZE,          \
+            ____size += snprintf(____info + ____size, AF_LOG_MSG_SIZE-____size, \
                                         " [AF_LOG_EXT] : ");                    \
-            ____size += snprintf(____info + ____size, AF_LOG_MSG_SIZE,          \
+            ____size += snprintf(____info + ____size, AF_LOG_MSG_SIZE-____size, \
                             ##__VA_ARGS__);                                     \
             if (____size == AF_LOG_MSG_SIZE) {                                  \
                 ____info[AF_LOG_MSG_SIZE - 1] = '\0';                           \
diff --git a/platforms/asterfusion-bf/cp2112_util/onie_tlvinfo.c b/platforms/asterfusion-bf/cp2112_util/onie_tlvinfo.c
index 85b20b4..088290e 100644
--- a/platforms/asterfusion-bf/cp2112_util/onie_tlvinfo.c
+++ b/platforms/asterfusion-bf/cp2112_util/onie_tlvinfo.c
@@ -795,7 +795,7 @@ bool tlvinfo_add_tlv (u_int8_t *eeprom, int tcode,
     tlvinfo_tlv_t *eeprom_tlv;
     int new_tlv_len = 0;
     u_int32_t value;
-    char data[MAX_TLV_VALUE_LEN];
+    char data[MAX_TLV_VALUE_LEN + 1];
     int eeprom_index;
     int max_size = TLV_TOTAL_LEN_MAX;
 
diff --git a/platforms/asterfusion-bf/src/bf_pltfm_chss_mgmt/bf_pltfm_bd_eeprom.c b/platforms/asterfusion-bf/src/bf_pltfm_chss_mgmt/bf_pltfm_bd_eeprom.c
index 705487f..05134a4 100644
--- a/platforms/asterfusion-bf/src/bf_pltfm_chss_mgmt/bf_pltfm_bd_eeprom.c
+++ b/platforms/asterfusion-bf/src/bf_pltfm_chss_mgmt/bf_pltfm_bd_eeprom.c
@@ -846,7 +846,7 @@ bf_pltfm_status_t bf_pltfm_bd_type_init()
             exit (0);
         } else {
             char tlv_str[lqe_valen + 1];
-            char tlv_header[128];
+            char tlv_header[lqe_valen + 1];
             char *p;
             while (fgets (tlv_str, lqe_valen, fp)) {
                 p = strchr(tlv_str, 0x0A);
diff --git a/platforms/asterfusion-bf/utils/xt-cfgen.sh b/platforms/asterfusion-bf/utils/xt-cfgen.sh
index d5d5064..b5df1e2 100755
--- a/platforms/asterfusion-bf/utils/xt-cfgen.sh
+++ b/platforms/asterfusion-bf/utils/xt-cfgen.sh
@@ -17,15 +17,13 @@ enable_uart=0
 enable_iic=1
 
 echo -e "${YELLOW}Notice: Start detecting and make sure that the switchd is not running${RES}"
-sleep 1
-install_bfnkdrv
 
 # Find uart_util which created by bsp and installed to $SDE_INSTALL/bin.
 # When found return its absolute path.
 # For a given X-T with different OS (such as ubuntu), the working tty may differ from /dev/ttyS1 on Debian.
 # Please help to test with all possible /dev/ttySx.
 # uart_util=`find / | grep uart_util`
-uart_util="$SDE_INSTALL/bin/uart_util"
+uart_util="$(type -p uart_util)"
 if [ ! -e $uart_util ]; then
     # For SONiC, uart_util is installed in /opt/bfn/install/bin
     uart_util="/opt/bfn/install/bin/uart_util"
@@ -375,12 +373,11 @@ fi
 echo "COMe     : $default_cme"
 
 if [ -f $cfgfile ]; then
-    echo ""
-    echo ""
-    echo -e "${RED} We found that the $cfgfile already exists and no new configuration file will be generated.${RES}"
-    echo -e "${RED} The purpose of this prompt is to take into account that your system may have worked fine in the past.${RES}"
-    echo -e "${RED} If you want to generate a new configuration file, delete the old one first.${RES}"
-    friendly_exit
+    if [ -f $cfgfile.lock ]; then
+        echo -e "${YELLOW}$cfgfile exists and is protected by $cfgfile.lock.${RES}"
+	exit 0
+    fi
+    echo -e "${YELLOW}Overwriting existing $cfgfile. Create $cfgfile.lock to prevent this.${RES}"
 fi
 
 echo ""
@@ -479,7 +476,7 @@ echo "# rpc-listen-point:127.0.0.1" >> $cfgfile
 echo "# rpc-listen-point:lo" >> $cfgfile
 echo "# rpc-listen-point:10.240.4.50" >> $cfgfile
 echo "# rpc-listen-point:ma1" >> $cfgfile
-echo "rpc-listen-point:127.0.0.1" >> $cfgfile
+echo "# rpc-listen-point:127.0.0.1" >> $cfgfile
 
 if [[ $xt_platform =~ "732" ]]; then
     echo "" >> $cfgfile
diff --git a/platforms/asterfusion-bf/utils/xt-setup.sh b/platforms/asterfusion-bf/utils/xt-setup.sh
index c2e77b7..4f570d3 100755
--- a/platforms/asterfusion-bf/utils/xt-setup.sh
+++ b/platforms/asterfusion-bf/utils/xt-setup.sh
@@ -22,7 +22,7 @@ friendly_exit()
 
 install_cgosdrv()
 {
-    cgosdrv="/lib/modules/`uname -r`/kernel/drivers/misc/cgosdrv.ko"
+    cgosdrv="${MODULES}/lib/modules/$(uname -r)/cgosdrv.ko"
     mod=`lsmod | grep cgosdrv`
     mod=${mod:0:7}
 
@@ -52,7 +52,7 @@ uninstall_cgosdrv()
 
 install_nct6779d()
 {
-    nct6779drv="/lib/modules/`uname -r`/kernel/drivers/misc/nct6779d.ko"
+    nct6779drv="${MODULES}/lib/modules/$(uname -r)/nct6779d.ko"
     mod=`lsmod | grep nct6779d`
     mod=${mod:0:8}
 
@@ -88,7 +88,7 @@ install_bfnkdrv()
         $SDE_INSTALL/bin/dma_setup.sh
     fi
 
-    bfnkdrv="$SDE_INSTALL/lib/modules/bf_kdrv.ko"
+    bfnkdrv="${MODULES}/lib/modules/bf_kdrv.ko"
     mod=`lsmod | grep bf_kdrv`
     mod=${mod:0:7}
 
